src/pydolphinscheduler/tasks/http.py

```diff
@@ -82,8 +106,27 @@ def __init__(
             raise PyDSParamException(
                 "Parameter http_method %s not support.", http_method
             )
+        
+        if isinstance(http_params, list):
+            warnings.warn(
+                "The `http_params` parameter currently accepts a dictionary instead of a list. Your parameter is being ignored.",
+                DeprecationWarning,
+            )
+            
         self.http_method = http_method
-        self.http_params = http_params or []
+        self.http_params = http_params  # Storing the original value
+
+    @property
+    def http_params(self):
+        """Property to convert http_params using ParameterHelper when accessed."""
+        return ParameterHelper.convert_params(self._http_params, direction=Direction.IN) if self._http_params else []
+
+    @http_params.setter
+    def http_params(self, value):
+        """Setter method for http_params."""
+        self._http_params = value
+
+
```

Comment:
remove because we directly use param `_http_params`
```suggestion
```

Reply:
@zhongjiajie  sir done